content/blog/2022/10/envvar-policy.md

```diff
@@ -0,0 +1,459 @@
+kind: ClusterAdmissionPolicy
+metadata:
+  name: environment-variable-policy
+spec:
+  module: ghcr.io/kubewarden/policies/environment-variable-policy:v0.1.1
+  mutating: false
+  rules:
+  - apiGroups: ["apps"]
+    apiVersions: ["v1"]
+    resources: ["deployments"]
+    operations:
+    - CREATE
+    - UPDATE
+  settings:
+    rules:
+      - reject: anyIn
+        environmentVariables:
+          - name: "foo"
+            value: "bar"
+          - name: "foo2"
+            value: "bar2"
+      - reject: allAreUsed
+        environmentVariables:
+          - name: "foo3"
+            value: "bar3"
+          - name: "foo4"
+            value: "bar4"
+EOF
+```
+
+The following two `Deployments` will be rejected. The first one will be rejected by the first rule:
+
+```console
+kubectl apply -f - <<EOF
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: nginx-deployment
+  labels:
+    app: nginx
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app: template-nginx
+  template:
+    metadata:
+      labels:
+        app: template-nginx
+    spec:
+      containers:
+      - name: nginx
+        image: nginx:1.14.2
+        ports:
+        - containerPort: 80
+        env:
+          - name: "foo"
+            value: "bar"
+EOF
+Error from server: error when creating "STDIN": admission webhook "clusterwide-environment-variable-policy.kubewarden.admission" denied the request: Resource cannot define any of the environment variables from the rule.
+```
+
+The second `Deployment`  will be reject by the second rule:
+
+```console
+kubectl apply -f - <<EOF
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: nginx-deployment
+  labels:
+    app: nginx
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app: template-nginx
+  template:
+    metadata:
+      labels:
+        app: template-nginx
+    spec:
+      containers:
+      - name: nginx
+        image: nginx:1.14.2
+        ports:
+        - containerPort: 80
+        env:
+          - name: "foo3"
+            value: "bar3"
+          - name: "foo4"
+            value: "bar4"
+EOF
+Error from server: error when creating "STDIN": admission webhook "clusterwide-environment-variable-policy.kubewarden.admission" denied the request: Resource cannot have all the environment variables from the rule defined.
+```
+
+But when we try to deploy a resource that respects both rules, it will be created with no issues:
+
+```console
+kubectl apply -f - <<EOF
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: nginx-deployment
+  labels:
+    app: nginx
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app: template-nginx
+  template:
+    metadata:
+      labels:
+        app: template-nginx
+    spec:
+      containers:
+      - name: nginx
+        image: nginx:1.14.2
+        ports:
+        - containerPort: 80
+        env:
+          - name: "foo3"
+            value: "bar3"
+EOF
+deployment.apps/nginx-deployment created
+```
+
+
+## A use case example
```

Comment:
As I've added a pretty complete example of the multiple rules in use. I decided to change this title as well. I did not remove the section completely because I think many examples help to understand what the policy is able of.

Reply:
I like it, thanks